iroha_network/src/lib.rs

```diff
@@ -1,17 +1,18 @@
+use futures::lock::Mutex;
```

Comment:
Now `futures::lock::Mutex` will also be used in sync functions, I am not sure if this is good for general design.

Reply:
Maybe introduce `SyncState` and `AsyncState` instead of `State`.

Reply:
@eadventurous maybe we can remove sync API? Where do we need it?

Reply:
Good point, right now nowhere, as from what I understand we decided to stick with async implementation for all our libraries.

Reply:
Will you remove it then?

Reply:
@eadventurous yes - I will touch `network` module a bit and also will delete it.